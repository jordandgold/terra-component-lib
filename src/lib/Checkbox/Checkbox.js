import React, { Component } from "react";
import "./Checkbox.scss";
import PropTypes from "prop-types";

class Checkbox extends Component {
  constructor(props) {
    super(props);

    this.state = {
      checked: this.props.checked || false
    };
  }

  handleChange = e => {
    this.setState({ checked: e.target.checked });
    this.props.handleChange(e);
  };

  render() {
    let isChecked = this.state.checked ? "is-checked" : "";

    return (
      <div className={`ter-checkbox ${isChecked}`}>
        <label className="ter-checkbox__label" htmlFor={this.props.name}>
          {this.props.label}
        </label>
        <input
          id={this.props.name}
          name={this.props.name}
          type="checkbox"
          checked={this.state.checked}
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

export default Checkbox;

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  handleChange: PropTypes.func.isRequired
};
