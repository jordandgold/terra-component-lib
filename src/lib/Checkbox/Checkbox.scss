@import "../global-styles/terra.scss";

.ter-checkbox {
  padding-top: spacing("xxs");
  padding-bottom: spacing("xxs");
  box-sizing: content-box;
  &__label {
    @include type-setting();
    font-size: $form-label-font-size;
    display: block;
    padding-bottom: spacing("mini");
    color: $form-label-color;
  }
  input[type="checkbox"] {
    opacity: 0;
    height: 0;
    width: 0;
    &:focus + label:before {
      outline: 0;
    }
  }
  &__label {
    @include type-setting();
    font-size: $form-label-font-size;
    position: relative;
    display: inline-block;
    padding-left: 22px;
    cursor: pointer;
    &:before,
    &:after {
      position: absolute;
      content: "";
      display: inline-block;
    }
    &:before {
      height: 12px;
      width: 12px;
      border: 1px solid $checkbox-border-color;
      left: 0px;
      top: 5px;
    }
    &:after {
      height: 8px;
      width: 8px;
      left: 3px;
      top: 8px;
      background: $checkbox-fill-color;
      transform: scale(0, 0);
      transition: transform 100ms ease-in-out;
      .is-checked & {
        transform: scale(1, 1);
      }
    }
  }
}
