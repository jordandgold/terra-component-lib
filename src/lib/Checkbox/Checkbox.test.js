import React from "react";
import { shallow } from "enzyme";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    wrapper = shallow(
      <Checkbox
        name="test name"
        label="test label"
        checked={false}
        handleChange={mockHandleChange}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange on click", () => {
    const mockEvent = {
      target: {
        name: "test name",
        checked: true
      }
    };
    wrapper.find("input").simulate("change", mockEvent);

    expect(mockHandleChange).toHaveBeenCalledWith(
      mockEvent.target.name,
      mockEvent.target.checked
    );
  });

  it("should set state", () => {
    const mockEvent = {
      target: {
        name: "test name",
        checked: true
      }
    };
    wrapper.find("input").simulate("change", mockEvent);

    expect(wrapper.state().checked).toEqual(true);
  });
});
