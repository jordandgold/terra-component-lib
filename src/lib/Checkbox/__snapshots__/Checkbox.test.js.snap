// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match the snapshot 1`] = `
<div
  className="ter-checkbox "
>
  <label
    className="ter-checkbox__label"
    htmlFor="test name"
  >
    test label
  </label>
  <input
    checked={false}
    id="test name"
    name="test name"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
